File: KITE-Hangouts-Test/js/steps/StartVideoCallStep.js
const {TestStep, TestUtils} = require('kite-common');
const waitAround = TestUtils.waitAround;

class StartVideoCallStep extends TestStep {

  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;
    this.timeout = kiteBaseTest.timeout;
    this.url = kiteBaseTest.url;
    this.uuid = kiteBaseTest.uuid;
    this.page = kiteBaseTest.page;
    this.id = kiteBaseTest.id;
    this.io = kiteBaseTest.io;
    this.usersPerRoom = kiteBaseTest.payload.usersPerRoom;
  }

  stepDescription() {
    if (this.id % this.usersPerRoom == 0) {
      return "Starting the video call";
    } else {
      return "Waiting for the video call to start";
    }
  }

  async step() {  
    if (this.id % this.usersPerRoom == 0) {
      await this.page.startVideoCall();
      await waitAround(1000);
      const roomUrl = await this.driver.getCurrentUrl();
      this.io.emit("registerRoomUrl", roomUrl, this.id);
      console.log("Video call started on url: " + roomUrl);
    } else {
      await waitAround(3 * 1000);
    } 
    await waitAround(2000); 
  }
}

module.exports = StartVideoCallStep;

File: KITE-Hangouts-Test/src/main/java/io/cosmosoftware/kite/hangouts/steps/StartVideoCallStep.java
package io.cosmosoftware.kite.hangouts.steps;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.hangouts.pages.MainPage;
import io.cosmosoftware.kite.manager.RoomManager;
import io.cosmosoftware.kite.steps.TestStep;
import org.webrtc.kite.tests.TestRunner;

import static io.cosmosoftware.kite.entities.Timeouts.ONE_SECOND_INTERVAL;
import static io.cosmosoftware.kite.entities.Timeouts.THREE_SECOND_INTERVAL;
import static io.cosmosoftware.kite.util.TestUtils.waitAround;

public class StartVideoCallStep extends TestStep {
  private final int id;
  private final MainPage mainPage;
  private final RoomManager roomManager;
  
  public StartVideoCallStep(TestRunner runner, RoomManager roomManager, MainPage mainPage) {
    super(runner);
    this.id = runner.getId();
    this.mainPage = mainPage;
    this.roomManager = roomManager;
  }

  @Override
  public String stepDescription() {
    if (id % roomManager.getUsersPerRoom() == 0) {
      return "Starting the video call";
    } else {
      return "Waiting for the video call to start";
    }
  }

  @Override
  protected void step() throws KiteTestException {
    if (id % roomManager.getUsersPerRoom() == 0) {
      mainPage.startVideoCall();
      waitAround(ONE_SECOND_INTERVAL);
      String roomUrl = webDriver.getCurrentUrl();
      roomManager.setDynamicUrl(id, roomUrl);
      logger.info("Video call started on url: " + roomUrl);
    } else {
      waitAround(THREE_SECOND_INTERVAL);
    }    
    
  }
  
}
